tests/reduce/test_async_reduce_err.py

```diff
@@ -109,71 +108,74 @@ async def start_server(udfs):
     await server.wait_for_termination()
 
 
-class TestAsyncReducerError(unittest.TestCase):
-    @classmethod
-    def setUpClass(cls) -> None:
-        global _loop
-        loop = asyncio.new_event_loop()
-        _loop = loop
-        _thread = threading.Thread(target=startup_callable, args=(loop,), daemon=True)
-        _thread.start()
-        udfs = NewAsyncReducer()
-        asyncio.run_coroutine_threadsafe(start_server(udfs), loop=loop)
-        while True:
-            try:
-                with grpc.insecure_channel("unix:///tmp/reduce_err.sock") as channel:
-                    f = grpc.channel_ready_future(channel)
-                    f.result(timeout=10)
-                    if f.done():
-                        break
-            except grpc.FutureTimeoutError as e:
-                LOGGER.error("error trying to connect to grpc server")
-                LOGGER.error(e)
-
-    @classmethod
-    def tearDownClass(cls) -> None:
-        try:
-            _loop.stop()
-            LOGGER.info("stopped the event loop")
-        except Exception as e:
-            LOGGER.error(e)
-
-    def test_reduce(self) -> None:
-        stub = self.__stub()
-        request, metadata = start_request(multiple_window=False)
-        generator_response = None
-        try:
-            generator_response = stub.ReduceFn(
-                request_iterator=request_generator(count=10, request=request)
-            )
-            counter = 0
-            for _ in generator_response:
-                counter += 1
-        except Exception as err:
-            self.assertTrue("Got a runtime error from reduce handler." in err.__str__())
-            return
-        self.fail("Expected an exception.")
-
-    def test_reduce_window_len(self) -> None:
-        stub = self.__stub()
-        request, metadata = start_request(multiple_window=True)
-        generator_response = None
-        try:
-            generator_response = stub.ReduceFn(
-                request_iterator=request_generator(count=10, request=request)
-            )
-            counter = 0
-            for _ in generator_response:
-                counter += 1
-        except Exception as err:
-            self.assertTrue(
-                "reduce create operation error: invalid number of windows" in err.__str__()
-            )
-            return
-        self.fail("Expected an exception.")
-
-    def __stub(self):
-        return reduce_pb2_grpc.ReduceStub(_channel)
+# TODO: Check why terminating even after mocking
```

Comment:
please fix this.

Reply:
We see a behavior where the target function does not bind to the mocked value. In our other test cases I used the same approach and it was working fine. I had tried different things, but wasn't helping so had commented this for now. 
I discussed this with @ab93 as well today. We would need to understand this further to see where is the problem stemming from.

Reply:
I think we have to figure that out before merging the PR, right?

Reply:
Ideally shouldn't be a blocker, manually have confirmed that the restarts occur as expected for both containers. This is only for UT.

Reply:
can you open a new issue to fix it and prioritize that?

Reply:
@vigith I have it working! Updated the mocks in some functions! 
The CI is passing now

I have still left the TODO there for myself to check and understand this a bit more deeper.